geoh5py/objects/grid2d.py

```diff
@@ -58,52 +59,55 @@ class Grid2D(GridObject):
     )
     _converter: type[Grid2DConversion] = Grid2DConversion
 
-    def __init__(self, object_type: ObjectType, **kwargs):
-        self._origin: np.ndarray = np.asarray(
-            tuple(np.zeros(3)), dtype=[("x", float), ("y", float), ("z", float)]
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        object_type: ObjectType,
+        u_cell_size: float = 1.0,
+        v_cell_size: float = 1.0,
+        u_count: int = 1,
+        v_count: int = 1,
+        vertical: bool = False,
+        dip: float = 0.0,
+        **kwargs,
+    ):
+        self._u_count: np.int32 = self.validate_count(u_count, "u")
+        self._v_count: np.int32 = self.validate_count(v_count, "v")
+
+        super().__init__(
+            object_type=object_type,
+            **kwargs,
         )
-        self._u_cell_size: float | None = None
-        self._v_cell_size: float | None = None
-        self._u_count: int | None = None
-        self._v_count: int | None = None
-        self._rotation: float = 0.0
-        self._vertical: bool = False
-        self._dip: float = 0.0
 
-        super().__init__(object_type, **kwargs)
+        self.u_cell_size = u_cell_size
+        self.v_cell_size = v_cell_size
+        self.dip = dip
+        self.vertical = vertical
 
     @property
-    def cell_center_u(self) -> np.ndarray | None:
+    def cell_center_u(self) -> np.ndarray:
```

Comment:
I agree with you, we could put cell_center_uv as deprecated, it's not very usefull as it's not considering origin nor rotation and dip